File: ee.py
import math

def calcular_excentricidade_elipse():
    """
    Calcula a excentricidade de uma elipse.
    Permite escolher entre inserir semieixo maior/menor ou semieixo maior/distância focal.
    """
    print("--- Calculadora de Excentricidade de Elipse ---")
    print("Selecione a opção de entrada de dados:")
    print("1. Semieixo maior (a) e Semieixo menor (b)")
    print("2. Semieixo maior (a) e Distância focal (c)")

    while True:
        opcao = input("Digite 1 ou 2: ")
        if opcao in ['1', '2']:
            break
        else:
            print("Opção inválida. Por favor, digite 1 ou 2.")

    try:
        if opcao == '1':
            a = float(input("Digite o valor do semieixo maior (a): "))
            b = float(input("Digite o valor do semieixo menor (b): "))

            if not (a > 0 and b > 0):
                raise ValueError("Os valores dos semieixos devem ser positivos.")
            if b >= a:
                raise ValueError("O semieixo menor (b) deve ser menor que o semieixo maior (a) para uma elipse.")

            c = math.sqrt(a**2 - b**2)
            excentricidade = c / a

        else: # opcao == '2'
            a = float(input("Digite o valor do semieixo maior (a): "))
            c = float(input("Digite o valor da distância focal (c): "))

            if not (a > 0 and c >= 0):
                raise ValueError("O semieixo maior deve ser positivo e a distância focal não negativa.")
            if c >= a:
                raise ValueError("A distância focal (c) deve ser menor que o semieixo maior (a) para uma elipse.")

            excentricidade = c / a

        print(f"\nO semieixo maior (a) é: {a:.4f}")
        if opcao == '1':
            print(f"O semieixo menor (b) é: {b:.4f}")
            print(f"A distância focal (c) calculada é: {c:.4f}")
        else:
            print(f"A distância focal (c) é: {c:.4f}")

        print(f"\nA excentricidade da elipse é: {excentricidade:.4f}")

    except ValueError as e:
        print(f"Erro: {e}")
    except Exception as e:
        print(f"Ocorreu um erro inesperado: {e}")

# Chama a função para executar o programa
if __name__ == "__main__":
    calcular_excentricidade_elipse()

File: ee.c
#include <stdio.h> // Para entrada e saída (printf, scanf)
#include <math.h>  // Para funções matemáticas (sqrt)
#include <stdlib.h> // Para a função exit()

int main() {
    int opcao;
    double a, b, c, excentricidade;

    printf("--- Calculadora de Excentricidade de Elipse ---\n");
    printf("Selecione a opcao de entrada de dados:\n");
    printf("1. Semieixo maior (a) e Semieixo menor (b)\n");
    printf("2. Semieixo maior (a) e Distancia focal (c)\n");

    while (1) { // Loop infinito para garantir uma entrada válida
        printf("Digite 1 ou 2: ");
        if (scanf("%d", &opcao) == 1 && (opcao == 1 || opcao == 2)) {
            break; // Sai do loop se a entrada for válida
        } else {
            printf("Opcao invalida. Por favor, digite 1 ou 2.\n");
            // Limpa o buffer de entrada para evitar loops infinitos com entradas inválidas
            while (getchar() != '\n');
        }
    }

    if (opcao == 1) {
        printf("Digite o valor do semieixo maior (a): ");
        while (scanf("%lf", &a) != 1 || a <= 0) {
            printf("Valor invalido. Digite um numero positivo para 'a': ");
            while (getchar() != '\n');
        }

        printf("Digite o valor do semieixo menor (b): ");
        while (scanf("%lf", &b) != 1 || b <= 0) {
            printf("Valor invalido. Digite um numero positivo para 'b': ");
            while (getchar() != '\n');
        }

        if (b >= a) {
            printf("Erro: O semieixo menor (b) deve ser menor que o semieixo maior (a) para uma elipse.\n");
            return 1; // Retorna com erro
        }

        c = sqrt(a*a - b*b);
        excentricidade = c / a;

    } else { // opcao == 2
        printf("Digite o valor do semieixo maior (a): ");
        while (scanf("%lf", &a) != 1 || a <= 0) {
            printf("Valor invalido. Digite um numero positivo para 'a': ");
            while (getchar() != '\n');
        }

        printf("Digite o valor da distancia focal (c): ");
        while (scanf("%lf", &c) != 1 || c < 0) {
            printf("Valor invalido. Digite um numero nao negativo para 'c': ");
            while (getchar() != '\n');
        }

        if (c >= a) {
            printf("Erro: A distancia focal (c) deve ser menor que o semieixo maior (a) para uma elipse.\n");
            return 1; // Retorna com erro
        }

        excentricidade = c / a;
    }

    printf("\nO semieixo maior (a) e: %.4lf\n", a);
    if (opcao == 1) {
        printf("O semieixo menor (b) e: %.4lf\n", b);
        printf("A distancia focal (c) calculada e: %.4lf\n", c);
    } else {
        printf("A distancia focal (c) e: %.4lf\n", c);
    }

    printf("\nA excentricidade da elipse e: %.4lf\n", excentricidade);

    return 0; // Indica que o programa executou com sucesso
}